Fix proc load

EmmyMiao87 opened the pull request:
Load which is belongs to LoadManger could not be presented in show proc "/jobs/dbId/load". This commit fix this bug.

EmmyMiao87 committed:
Add streaming job in LoadProc

EmmyMiao87 committed:
Add the desired size of load scheduler queue

1. Create load stmt will be rejected when queue is full.
2. Desired max size is only a expect number, so the size of queue may be more then this number sometimes.

EmmyMiao87 committed:
Merge load manager and load jobs in jobs proc dir

EmmyMiao87 commented:
The latest commit is not belongs to this branch, so I reset hard the commit and forced pushed.

morningman commented on fe/src/main/java/org/apache/doris/load/loadv2/LoadJobScheduler.java:
```diff
@@ -45,6 +45,8 @@
 
     private static final Logger LOG = LogManager.getLogger(LoadJobScheduler.class);
+
+    private static final int desiredMaxWaitingSize = 100;
```
Change it to a configuration of frontend

morningman commented on fe/src/main/java/org/apache/doris/load/loadv2/LoadManager.java:
```diff
@@ -90,6 +90,9 @@ public void createLoadJobFromStmt(LoadStmt stmt) throws DdlException {
                 throw new DdlException("LoadManager only support the broker load.");
             }
+            if (loadJobScheduler.isQueueFull()) {
+                throw new DdlException("There are too many load job in waiting queue, please retry later.");
```
```suggestion
                throw new DdlException("There are more than " + Config.max_pending_load_jobs + " load jobs in waiting queue, please retry later.");
```

morningman commented on fe/src/main/java/org/apache/doris/load/loadv2/LoadJobScheduler.java:
```diff
@@ -88,6 +90,10 @@ private void process() throws InterruptedException {
     }
 
+    public boolean isQueueFull() {
+        return needScheduleJobs.size() > desiredMaxWaitingSize;
```
```suggestion
        return needScheduleJobs.size() > Config.max_pending_load_jobs;
```

morningman commented on fe/src/main/java/org/apache/doris/load/loadv2/LoadManager.java:
```diff
@@ -314,6 +332,12 @@ public void processTimeoutJobs() {
 
+    // the meta version info of load could not be presented
+    public List<List<Comparable>> getLoadJobUnfinishedInfo(long jobId) {
+        LinkedList<List<Comparable>> infos = new LinkedList<List<Comparable>>();
```
What's this for?

EmmyMiao87 replied:
For show something later？

morningman commented on fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java:
```diff
@@ -57,6 +57,7 @@
 public class RoutineLoadManager implements Writable {
     private static final Logger LOG = LogManager.getLogger(RoutineLoadManager.class);
     private static final int DEFAULT_BE_CONCURRENT_TASK_NUM = 10;
+    private static final int desiredMaxWaitingSize = 100;
```
Change it config, can be same as max_pending_load_jobs

morningman commented on fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java:
```diff
@@ -148,9 +149,11 @@ public void addRoutineLoadJob(RoutineLoadJob routineLoadJob, String dbName) thro
                         + dbName);
             }
+            if (getRoutineLoadJobByState(RoutineLoadJob.JobState.NEED_SCHEDULE).size() > desiredMaxWaitingSize) {
+                throw new DdlException("There are too many routine load job in waiting queue, please retry later");
```
```suggestion
                throw new DdlException("There are more than " + Config.max_pending_load_jobs + " routine load job in waiting queue, please retry later");
```

EmmyMiao87 committed:
Add a config named desired_max_waiting_jobs

1. If the number of pending load jobs is more then desired_max_waiting_jobs, the create load stmt will be rejected.
2. If the number of need_scheduler load jobs is more then desired_max_waiting_jobs, the new routine load job will be rejected.

EmmyMiao87 committed:
Forget to add file

morningman reviewed (approved):
LGTM
